onnxruntime/core/providers/cpu/ml/svmclassifier.h

```diff
@@ -29,34 +31,98 @@ class SVMCommon {
   void set_kernel_type(KERNEL new_kernel_type) { kernel_type_ = new_kernel_type; }
   KERNEL get_kernel_type() const { return kernel_type_; }
 
+  void batched_kernel_dot(const gsl::span<const T> a, const gsl::span<const T> b,
+                          int64_t m, int64_t n, int64_t k,
+                          float scalar_C,
+                          const gsl::span<T> out,
+                          concurrency::ThreadPool* threadpool) const {
+    assert(a.size() == size_t(m * k) && b.size() == size_t(k * n) && out.size() == size_t(m * n));
+
+    if (kernel_type_ == KERNEL::RBF) {
+      T* cur_out = out.data();
+      const T* cur_batch = a.data();
+
+      // each batch has 'k' features
+      for (int64_t batch = 0; batch < m; ++batch) {
+        const T* cur_support_vector = b.data();
+
+        // broadcast the support vectors against the k features in each batch. output is one value per support vector
+        for (int64_t support_vector = 0; support_vector < n; ++support_vector) {
+          T sum = 0.f;
+          const T* cur_input = cur_batch;
+
+          for (int64_t feature = 0; feature < k; ++feature) {
+            T val = *cur_input++ - *cur_support_vector++;
+            sum += val * val;
+          }
+
+          *cur_out++ = std::exp(-gamma_ * sum);
+        }
+
+        cur_batch += k;  // move to start of next batch
+      }
+    } else {
+      float alpha = 1.f;
+      float beta = 1.f;
+      static const TensorShape shape_C({1});
+      float c = scalar_C;  // scalar_C is used for LINEAR in the GEMM
+
+      if (kernel_type_ != KERNEL::LINEAR) {
+        // kernel_type_ == POLY or SIGMOID
+        alpha = gamma_;
+        c = coef0_;
+      }
+
+      onnxruntime::Gemm<T>::ComputeGemm(CBLAS_TRANSPOSE::CblasNoTrans, CBLAS_TRANSPOSE::CblasTrans,
+                                        m, n, k,
+                                        alpha, a.data(), b.data(), beta,
+                                        c != 0.f ? &c : nullptr, &shape_C,
+                                        out.data(),
+                                        threadpool);
+
+      if (kernel_type_ == KERNEL::POLY) {
+        auto map_out = EigenVectorArrayMap<T>(out.data(), out.size());
+        if (degree_ == 2)
```

Comment:
switch? the performance gain will probably be none since degree_ is usually 2.

Reply:
I'll leave it for now so I don't need to wait on CI rebuilding everything and getting another sign off. could be changed to a switch in the future but I agree you won't see much perf diff compared to the cost of all the other operations.